### DISCUSSION TITLE
AdonisJs 5 Build Error can't run the production build.

### DISCUSSION THREAD

**Discussion by dev-techguy** [adonisjs/core#2806] [Help]
![Screenshot from 2021-06-17 09-58-52](https://user-images.githubusercontent.com/36882600/122350507-f1c01800-cf55-11eb-9c3e-a88743b8ff36.png)

**Comment by dev-techguy** [comment]
This the.env file


"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Env_1 = __importDefault(global[Symbol.for('ioc.use')]("Adonis/Core/Env"));
exports.default = Env_1.default.rules({
    HOST: Env_1.default.schema.string({ format: 'host' }),
    PORT: Env_1.default.schema.number(),
    APP_KEY: Env_1.default.schema.string(),
    APP_NAME: Env_1.default.schema.string(),
    NODE_ENV: Env_1.default.schema.enum(['development', 'production', 'testing']),
});
//# sourceMappingURL=env.js.map

**Comment by RomainLanz** [comment]
Hey @dev-techguy! 👋 

Have you even read the error message?
It tells you that you are missing some environment variables.

**Comment by maswebber** [comment]
ah this a bug, you can just copy the` .env ` file from root project and paste it in build folder

**Reply by thetutlage** [reply]
Read this https://docs.adonisjs.com/guides/environment-variables#do-not-commit-the-env-file. Infact the entire doc

**Comment by sreesdas** [comment]
Go to the Build folder and execute the command:

ENV_PATH=../.env node server.js

Basically you need to set the node env variables as the scripts expects these

**Reply by sreesdas** [reply]
A good practice is to NOT store the env file along with the source code and keep it somewhere else.
In a production setup, tools like pm2 has their own way of setting env variables.
